internal/rbd/controllerserver.go

```diff
@@ -588,6 +596,14 @@ func (cs *ControllerServer) createVolumeFromSnapshot(
 
 	log.DebugLog(ctx, "create volume %s from snapshot %s", rbdVol.RequestName, rbdSnap.RbdSnapName)
 
+	// resize the volume if the size is different
+	if rbdSnap.SizeBytes != rbdVol.VolSize {
+		err = rbdVol.resize(rbdVol.VolSize)
```

Comment:
Reducing size is dangerous, that should not be done.

Consider placing these resize checks/actions in a helper function:

```go
err = rbdVol.Expand(rbdSnap.SizeBytes)
...

// Expand resizes the rbdImage to the given size, in case the rbdImage is smaller.
func (ri *rbdImage) Expand(size int) error {
    if ri.VolSize >= size {
         return nil
    }

    return ri.resize(size)
}
```

Reply:
the smaller size is not allowed for now in kubernetes  (CO). will send a follow-up patch to introduce this check for cephcsi (for both cephfs and rbd) to disallow that operation as it might cause some problems like data loss etc.

Reply:
Indeed **NEVER** reduce the size of an RBD volume.

Changing quotas for CephFS would be acceptable, in case the current consumption does not hit the limit.

Reply:
https://github.com/ceph/ceph-csi/issues/2718

Reply:
will do some testing for cephfs how it works the reducing size of a bigger size subvolume and will send follow-up for this one. opened #2718 for tracking it.

Reply:
moved to a helper function